lib/search_parser.rb

```diff
@@ -0,0 +1,62 @@
+require 'parslet'
+
+class SearchParser < Parslet::Parser
+  rule(:spaces) { match('\s').repeat(1) }
+  rule(:spaces?) { spaces.maybe }
+
+  rule(:quoted_string) { double_quoted_string | single_quoted_string }
+  rule(:double_quoted_string) {
+    str('"') >> (
+      str('"').absent? >> any
+    ).repeat.as(:string) >> str('"')
+  }
+  rule(:single_quoted_string) {
+    str("'") >> (
+      str("'").absent? >> any
+    ).repeat.as(:string) >> str("'")
+  }
+
+  rule(:bare_string) {
+    match('[!\w-]').repeat(1).as(:string)
+  }
+  rule(:string) { quoted_string | bare_string }
+
+  rule(:regex) { # /(((\\\/)|\\)[^\/])*/
+    str('/') >> (
+      (str('\\/') |
+      str('\\')) |
+      match('[^/]')
+    ).repeat.as(:regex) >> str('/')
+  }
+
+  # Note that while this list should generally be kept sorted, an entry that is a prefix of
```

Comment:
this comment is out of place here now and probably needs to be with the field and keyword definitions.

Reply:
Actually now I think about it I don't think this is an issue any more since keywords aren't validated at the parsing stage any more.